compat/memstream.c

```diff
@@ -0,0 +1,30 @@
+/* vi:set ts=4 sw=4 expandtab:
+*
+* Copyright 2016, Chris Leishman (http://github.com/cleishm)
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#include "../config.h"
+#include <errno.h>
+#include <limits.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <fmem.h>
+
+fmem fm;
+
+FILE* open_memstream(char** mem, size_t* size)
```

Comment:
Rather than a separate file in a `compat` directory, can this be added to `lib/test/memstream.c` using `#ifdef` directives to optionally include?

Reply:
Yes